### ISSUE TITLE
Starting with Sphinx 1.5.2, project docs error out with "make clean html"

### ISSUE THREAD

**Issue by bryevdv** [sphinx-doc/sphinx#3731]
Subject: <what happen when you do on which document project>

### Problem

The Bokeh project supprts python 2.7 and 3.4+, and also predates python Enum classes. Accordingly, we have our own "enums" objects, and these appear to be confusing recent versions of Sphinx. 

Full output below, the relevant parts appear to be these:
```
[autodoc] /Users/bryan/work/bokeh/sphinx/source/docs/reference/core/enums.rst:23: input:
.. automodule:: bokeh.core.enums
  :members:
  :undoc-members:
  :exclude-members: enumeration
[autodoc] import bokeh.core.enums
[autodoc] => <module 'bokeh.core.enums' from '/Users/bryan/work/bokeh/bokeh/core/enums.py'>
[autodoc] from bokeh.core.enums import Aggregation
[autodoc] import bokeh.core.enums
[autodoc] => <module 'bokeh.core.enums' from '/Users/bryan/work/bokeh/bokeh/core/enums.py'>
[autodoc] getattr(_, 'Aggregation')
[autodoc] => Enumeration(sum, mean, count, nunique, median, min, max)
```
This results in
```
  File "/Users/bryan/anaconda/lib/python3.6/site-packages/Sphinx-1.5.4-py3.6.egg/sphinx/ext/autodoc.py", line 780, in get_object_members
    if isenumclass(self.object):
  File "/Users/bryan/anaconda/lib/python3.6/site-packages/Sphinx-1.5.4-py3.6.egg/sphinx/util/inspect.py", line 107, in isenumclass
    return issubclass(x, enum.Enum)
TypeError: issubclass() arg 1 must be a class
```

Our project enums are *objects*, not classes. My best guess is that Sphinx is assuming they they are python enums for some reason. I would suggest that the `isenumclass` (and possibly others) should protect against this, and *not* assume that things passed are classes (this should be checked explicitly). 

In the mean time, **is there a way to explicitly tell Sphinx that our poject enums are not python enums?**

#### Procedure to reproduce the problem
gist output attached below was generated from the Bokeh docs with the command:
```
BOKEH_DOCS_CDN=local make SPHINXOPTS=-vv clean html```
```

#### Error logs / results

https://gist.github.com/bryevdv/49104384a5d3c8e074d4c085072dd5df

#### Expected results

Docs should build without error or warning, as is the case with Sphinx 1.5.1

### Reproducible project / your project

https://github.com/bokeh/bokeh
http://bokeh.pydata.org/en/latest/docs/dev_guide/documentation.html

### Environment info
- OS: OSX
- Python version: 3.6 
- Sphinx version: 1.5.2+

**Comment by bryevdv** [comment]
The following new definition in `sphinx/util/inspect.py` allow our docs to build successfully:

```
def isenumclass(x):
    """Check if the object is subclass of enum."""
    if enum is None:
        return False
    return inspect.isclass(x) and issubclass(x, enum.Enum)
```

I will submit a PR but I am still very interested to know if there is any way to explicitly work around this in with existing Sphinx versions.

**Comment by tk0miya** [comment]
Now this was already fixed by #3732. So I close this.
Thanks!
